buildpipeline/pipeline.json

```diff
@@ -7,632 +7,335 @@
           "Parameters": {
             "PB_DockerTag": "centos-7-d485f41-20173404063424",
-            "PB_BuildArguments": "-buildArch=x64 -Release -stripSymbols",
-            "PB_BuildTestsArguments": "-buildArch=x64 -Release -SkipTests -Outerloop -- /p:ArchiveTests=true /p:EnableDumpling=true",
+            "PB_BuildArguments": "-buildArch=x64 -$(PB_ConfigurationGroup) -stripSymbols",
+            "PB_BuildTestsArguments": "-buildArch=x64 -$(PB_ConfigurationGroup) -SkipTests -Outerloop -- /p:ArchiveTests=true /p:EnableDumpling=true",
             "PB_SyncArguments": "-p -- /p:ArchGroup=x64 /p:DotNetRestoreSources=$(PB_RestoreSource)",
             "PB_TargetQueue": "Centos.73.Amd64+RedHat.72.Amd64+RedHat.73.Amd64+Debian.87.Amd64+Debian.90.Amd64+Ubuntu.1404.Amd64+Ubuntu.1604.Amd64+Ubuntu.1704.Amd64+Ubuntu.1710.Amd64+suse.422.amd64+SLES.12.Amd64+fedora.25.amd64+Fedora.26.Amd64",
-            "PB_CreateHelixArguments": "/p:ArchGroup=x64 /p:ConfigurationGroup=Release /p:TestProduct=corefx /p:TimeoutInSeconds=1200 /p:TargetOS=Linux"
+            "PB_CreateHelixArguments": "/p:ArchGroup=x64 /p:ConfigurationGroup=$(PB_ConfigurationGroup) /p:TestProduct=corefx /p:TimeoutInSeconds=1200 /p:TargetOS=Linux"
           },
           "ReportingParameters": {
             "OperatingSystem": "Linux",
             "Platform": "x64",
-            "Type": "build/product/",
-            "ConfigurationGroup": "Release"
+            "Type": "build/product/"
           }
         },
         {
           "Name": "DotNet-CoreFx-Trusted-Linux",
           "Parameters": {
             "PB_DockerTag": "centos-6-376e1a3-20174311014331",
-            "PB_BuildArguments": "-buildArch=x64 -Release -stripSymbols -RuntimeOS=rhel.6 -- /p:PortableBuild=false",
-            "PB_BuildTestsArguments": "-buildArch=x64 -Release -SkipTests -Outerloop -RuntimeOS=rhel.6 -- /p:ArchiveTests=true /p:EnableDumpling=true /p:PortableBuild=false",
+            "PB_BuildArguments": "-buildArch=x64 -$(PB_ConfigurationGroup) -stripSymbols -RuntimeOS=rhel.6 -- /p:PortableBuild=false",
+            "PB_BuildTestsArguments": "-buildArch=x64 -$(PB_ConfigurationGroup) -SkipTests -Outerloop -RuntimeOS=rhel.6 -- /p:ArchiveTests=true /p:EnableDumpling=true /p:PortableBuild=false",
             "PB_SyncArguments": "-p -RuntimeOS=rhel.6 -- /p:ArchGroup=x64 /p:PortableBuild=false /p:DotNetRestoreSources=$(PB_RestoreSource)",
             "PB_TargetQueue": "RedHat.69.Amd64",
-            "PB_CreateHelixArguments": "/p:ArchGroup=x64 /p:ConfigurationGroup=Release /p:TestProduct=corefx /p:TimeoutInSeconds=1200 /p:TargetOS=Linux"
+            "PB_CreateHelixArguments": "/p:ArchGroup=x64 /p:ConfigurationGroup=$(PB_ConfigurationGroup) /p:TestProduct=corefx /p:TimeoutInSeconds=1200 /p:TargetOS=Linux"
           },
           "ReportingParameters": {
             "OperatingSystem": "RedHat6",
             "Platform": "x64",
-            "Type": "build/product/",
-            "ConfigurationGroup": "Release"
```

Comment:
This is only for Mission Control reporting and will likely mess things up if you remove it.  We may be able to get away with simply moving the view configuration to point to PB_ConfigurationGroup assuming it's always supplied, I'd have to check with @ChadNedzlek .  I'd recommend just skipping this line for now, to prevent reporting weirdness.

Reply:
My expectation is that it will simply not have a ConfigurationGroup but if it does cause issues I will update the view in mission control accordingly.

Reply:
It will cause some history to get a bit confused, since we'll go from "Failed in release" to "failed in null".  Changing the view configuration would similarly confuse the past, since we would lose the ability to differentiate the past runs. The safest thing to do it leave it (in case we ever want to add debug back in the future, especially)

Reply:
I'm planning to use the same definitions to queue Release and Debug builds intertwined. Is there a way to use a VSTS parameter in this property and always pass in PB_ConfigurationGroup to this parameter?

Reply:
Nothing we have at the moment, unfortunately.  We hadn't considered this scenario (where we want to change the name of a parameter but have it still mean the same thing) when we rather hastily built the reporting framework.

Reply:
I'll discuss this with @ChadNedzlek when he gets in.  While he's right about how Mission Control's view configuration works, I don't think this is an unsolveable problem. 

Rather, we could introduce the following support fairly easy to Pipebuild itself:

```
           "ReportingParameters": {
             "OperatingSystem": "RedHat6",
             "Platform": "x64",
            "Type": "build/product/",
            "ConfigurationGroup": "$(PB_ConfigurationGroup)"
```
Then, when sending the reporting telemetry from Pipebuild itself we'd simply use a regex to match any $()s we see, and if they match the name of a variable used while queueing that specific build, we substitute that variable's value.    Folks will see this in the definitions and be able to do it.

Until something like that is in, I still favor just leaving this alone;  it's still true (the builds are release) and leaves us time to get a reasonable fix in.